src/gui/MainWindow.cc

```diff
@@ -2297,12 +2303,138 @@ void MainWindow::actionRenderDone(const shared_ptr<const Geometry>& root_geom)
   compileEnded();
 }
 
+
+void MainWindow::actionMeasureDistance()
+{
+	this->qglview->selected_obj.clear();
+	this->qglview->update();
+	this->qglview->measure_state=MEASURE_DIST1;
+}
+
+void MainWindow::actionMeasureAngle()
+{
+	this->qglview->selected_obj.clear();
+	this->qglview->update();
+	this->qglview->measure_state=MEASURE_ANG1;
+}
+
+extern double calculateLineLineDistance(const Vector3d &l1b, const Vector3d &l1e, const Vector3d &l2b, const Vector3d &l2e, double &dist_lat);
+extern double calculateLinePointDistance(const Vector3d &l1, const Vector3d &l2, const Vector3d &pt, double & dist_lat) ;
+
+void MainWindow::leftClick(QPoint mouse) 
```

Comment:
Can we get all the measurement code into a separate class? MainWindow is already way too large.